Documentation/Evolution/CharacterClasses.md

```diff
@@ -0,0 +1,354 @@
+- `Lowercase_Letter`
+- `Titlecase_Letter`
+- `Other_Letter`
+- `Decimal_Number`
+
+| Matching Level | Note |
+|-------------------------------|---------|
+| Character | any character where the general category of the first unicode scalar is in the above list |
+| Unicode Scalar | any scalar where the general category is in the above list |
+| POSIX | `[A-Za-z0-9_]` |
+
+`\W` matches the inverse set of characters/elements.
+
+**DSL:** The corresponding pattern is called `.wordCharacter`:
+
+```swift
+Pattern {
+    OneOrMore(.wordCharacter)
+}
+```
+
+
+### Whitespace and Newlines: `\s`, `\S` (plus `\h`, `\H`, `\v`, `\V`, and `\R`)
+
+`\s` matches any single, non-zero width whitespace character, as denoted by the following list.
+
+- `CHARACTER TABULATION` (U+0009)
+- `LINE FEED (LF)` (U+000A)*
+- `LINE TABULATION` (U+000B)*
+- `FORM FEED (FF)` (U+000C)*
+- `CARRIAGE RETURN (CR)` (U+000D)*
+- `NEWLINE (NEL)` (U+0085)*
+- any character in the Unicode general category `Z`/`Separator`
+- the CR/LF sequence
+
+`\h` matches the tab character (U+0009) or any character in the Unicode general category `Space_Separator`.
+
+`\v` matches the five line separator characters above, marked with *, as well as (U+2028) and (U+2029), not including the CR/LF sequence.
+
+`\R` matches the same characters as `\v`, and in addition matches the CR/LF sequence.
+
+| Matching level | Note |
+|-------------------------------|---------|
+| Character | any character where the general category of the first unicode scalar is in the above list |
+| Unicode Scalar | any scalar where the general category is in the above list |
+| POSIX | `[ \t\n\r\f\v]` |
+
+`\S`, `\H`, and `\V` match the inverse set of characters as their lowercase respective character classes.
+
+**DSL:** The corresponding pattern is called `.whitespaceOrNewline`. Additional patterns for `.whitespace` and `.newline` are also available.
+
+```swift
+Pattern {
+    "abc"
+    OneOrMore(.whitespace)
+    "def"
+    OneOrMore(.whitespaceOrNewline)
+}
+```
+
+### Control characters: `\t`, `\r`, `\n`, `\f`, `\0`, `\e`, `\a`, `\b`, `\cX`
+
+These escaped literal characters represent specific control characters only.
+
+- `\t`: `CHARACTER TABULATION` (U+0009)
+- `\r`: `CARRIAGE RETURN (CR)` (U+000D)
+- `\n`: `LINE FEED (LF)` (U+000A)
+- `\f`: `FORM FEED (FF)` (U+000C)
+- `\0`: `NUL` (U+0000)
+- `\e`: `ESCAPE` (U+001B)
+- `\a`: `BELL` (U+0007)
+- `\b`: `BACKSPACE` (U+0008) (within a character set only)
+- `\cX`: The control character indicated by Control-`X`; in the code point range `0..<32`.
+
+**DSL:** The corresponding patterns are available as named patterns, or alternatively as character literals.
+
+- `\t`: `.tabCharacter` or `"\u{09}"`
+- `\r`: `.carriageReturn` or `"\u{0d}"`
+- `\n`: `.lineFeed` or `"\u{0a}"`
+- `\f`: `.formFeed` or `"\u{0c}"`
+- `\0`: `.nul` or `"\u{0}"`
+- `\e`: `.escape` or `"\u{1b}"`
+- `\a`: `.bell` or `"\u{07}"`
+- `\b`: `.backspace` or `"\u{08}"`
+- `\cX`: `.control(_: UnicodeScalar)`
+
```

Comment:
Is `\n` just LF, or is this configurable? How does this relate to `Character.isNewline`?